wallet-test/src/test/scala/org/bitcoins/wallet/models/DLCDAOTest.scala

```diff
@@ -0,0 +1,199 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.config.RegTest
+import org.bitcoins.core.currency.Satoshis
+import org.bitcoins.core.hd.{BIP32Path, HDAccount}
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.EmptyScriptPubKey
+import org.bitcoins.core.protocol.transaction.{
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.dlc.DLCMessage
+import org.bitcoins.testkit.fixtures.WalletDAOFixture
+import org.bitcoins.testkit.wallet.{BitcoinSWalletTest, DLCWalletTestUtil}
+
+class DLCDAOTest extends BitcoinSWalletTest with WalletDAOFixture {
+
+  behavior of "DLCDAO"
+
+  it should "correctly insert a DLC into the database" in { daos =>
+    val dlcDAO = daos.dlcDAO
+
+    val eventId = DLCWalletTestUtil.sampleDLCEventId
+
+    val dlcDb = DLCDb(
+      eventId = eventId,
+      isInitiator = true,
+      account = HDAccount.fromPath(BIP32Path.fromString("m/84'/0'/0'")).get,
+      keyIndex = 0,
+      winSigOpt = None,
+      loseSigOpt = None,
+      refundSigOpt = None,
+      oracleSigOpt = Some(DLCWalletTestUtil.sampleOracleLoseSig)
+    )
+
+    for {
+      _ <- dlcDAO.create(dlcDb)
+
+      readDLC <- dlcDAO.read(eventId)
+    } yield {
+      assert(readDLC.contains(dlcDb))
+    }
+  }
+
+  it should "correctly insert a DLCOffer into the database" in { daos =>
+    val dlcDAO = daos.dlcDAO
+    val offerDAO = daos.dlcOfferDAO
+
+    val offer = DLCWalletTestUtil.sampleDLCOffer
+    val eventId = DLCMessage.calcEventId(offer.oracleInfo,
+                                         offer.contractInfo,
+                                         offer.timeouts)
+
+    for {
+      _ <- {
+        val dlcDb = DLCDb(
+          eventId = eventId,
+          isInitiator = true,
+          account = HDAccount.fromPath(BIP32Path.fromString("m/84'/0'/0'")).get,
+          keyIndex = 0,
+          winSigOpt = None,
+          loseSigOpt = None,
+          refundSigOpt = None,
+          oracleSigOpt = None
+        )
+        dlcDAO.create(dlcDb)
+      }
+      _ <- {
+        val offerDb = DLCOfferDb.fromDLCOffer(offer, RegTest)
+        offerDAO.create(offerDb)
+      }
+
+      readOffer <- offerDAO.read(eventId)
+    } yield {
+      assert(
+        readOffer.get.toDLCOffer(offer.fundingInputs).toJson == offer.toJson)
+    }
+  }
+
+  it should "correctly insert a DLCAccept into the database" in { daos =>
```

Comment:
the `it should "correctly insert a *** into the database` tests seem to follow a similar pattern and I think that we can factor it out to being a function that takes in an element, a DAO capabale of writing and reading that kind of element, and returning a `Future[Assertion]`